### DISCUSSION TITLE
Newbie with a question and some possible requests

### DISCUSSION THREAD

**Discussion by ithanium2** [BartoszCichecki/LenovoLegionToolkit#805] [Q&A] [answered]
Hello,
First of all, I want to thank you for this superb utility. I am a new user of Legion 5 Pro 16ARH7H model 82RG. I am new to AMD CPU's and to Lenovo laptops (used to have Asus and Clevo).
1. Since we have multiple presets for custom mode, is it possible to allow us to assign a color to them and change the power button LED color accordingly?
2. In custom mode, is it possible to let us set the CPU Temperature Limit lower then 85 degrees? Perhaps to 75 or 80. Also for the Peak Power Limit slider (I wanted to create a 30W TDP profile but slider minimum value is 35).
3. Regarding the Fan curve, I assume that it's somehow fixed value, because when we load from Quiet, Balanced or Performance only the TDP values change. I am asking this because I think that Lenovo has different fan curve for each profile (I might be mistaken).
4. Is it possible to add a CPU undervolting feature? (I know that there are some other apps out there that do that, but my CPU Ryzen 7 6800H is not supported)
5. I know we have TDP limits, but is it possible to add a feature that allows us to limit the actual frequency of the CPU? I know that there is something in the Windows Power Plans settings, but for some reason it does not seem to work all the time (I tested in Forza Horizon 5 and even tho I set the maximum processor frequency to 4GHz, it was going up to 4.5GHz for some reason).

**Comment by BartoszCichecki** [comment, marked as answer]
1. Not possible
2. U can use the minimum offset at the bottom of the page and set lower values. It is not guaranteed to work, because it will be technically out of spec value, but you can experiment yourself.
3. There are different fan curves for Quiet, Balance and Performance which are not modifiable.
4. There are other tools for that. If they don't work, I can't help.
5. The limits in power plan settings are the way to go, maybe some other apps can do this too.

**Comment by ithanium2** [comment]
Thank you for your quick response.

2. Well great, now I know the role of those offset settings :)  For temperatures I know for sure it works, because I tested it with UxTU.
3. Here, perhaps you misunderstood me. I do not want to modify the existing presets fan curves, but I was wondering if you have access to read them and apply them to the custom mode settings, in the same manner you apply the TDP values from the default profiles.
4. Fair enough. The tool is great as it is, I was just wondering if you were to consider adding voltage controls in the future.
5. Yes, I know that the power plan settings should work, I need to do some more testing in other games and such...maybe there is some anomaly with FH5.

**Comment by BartoszCichecki** [comment]
2. Oh, I see. That info is stored on gen 7 and newer devices, so depending on which model you have it might be there.

I actually don't know why I didn't add it to LLT, because I should and I will put it in next version :)

EDIT: Ah, I actually know why: It was because this data is not correct. It contains the default fan table for some reason.
